ionoscloud/data_source_networkloadbalancer_forwardingrule.go

```diff
@@ -178,28 +177,34 @@ func dataSourceNetworkLoadBalancerForwardingRuleRead(ctx context.Context, d *sch
 		}
 	} else {
 		/* search by name */
-		networkLoadBalancerForwardingRules, apiResponse, err := client.NetworkLoadBalancersApi.DatacentersNetworkloadbalancersForwardingrulesGet(ctx, datacenterId.(string), networkloadbalancerId.(string)).Depth(1).Filter("name", name.(string)).Execute()
+		var networkLoadBalancerForwardingRules ionoscloud.NetworkLoadBalancerForwardingRules
+
+		networkLoadBalancerForwardingRules, apiResponse, err := client.NetworkLoadBalancersApi.DatacentersNetworkloadbalancersForwardingrulesGet(ctx, datacenterId.(string), networkloadbalancerId.(string)).Execute()
 		logApiRequestTime(apiResponse)
 		if err != nil {
 			return diag.FromErr(fmt.Errorf("an error occurred while fetching network loadbalancers forwarding rules: %s", err.Error()))
 		}
 
-		if networkLoadBalancerForwardingRules.Items != nil && len(*networkLoadBalancerForwardingRules.Items) > 0 {
-			networkLoadBalancerForwardingRule = (*networkLoadBalancerForwardingRules.Items)[len(*networkLoadBalancerForwardingRules.Items)-1]
-			log.Printf("[INFO] %v network load balancers forwarding rules found matching the search criteria. Getting the latest network load balancer forwarding rule from the list %v", len(*networkLoadBalancerForwardingRules.Items), *networkLoadBalancerForwardingRule.Id)
-		} else {
-			return diag.FromErr(fmt.Errorf("no network load balancer forwarding rule found with the specified name %s", name.(string)))
+		var results []ionoscloud.NetworkLoadBalancerForwardingRule
+		if networkLoadBalancerForwardingRules.Items != nil {
+			for _, c := range *networkLoadBalancerForwardingRules.Items {
+				tmpNetworkLoadBalancerForwardingRule, apiResponse, err := client.NetworkLoadBalancersApi.DatacentersNetworkloadbalancersForwardingrulesFindByForwardingRuleId(ctx, datacenterId.(string), networkloadbalancerId.(string), *c.Id).Execute()
```

Comment:
let's do like for server, first check name == properties.name and only if they're equal findById